packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -1105,26 +1150,24 @@ class PopupMenuButtonState<T> extends State<PopupMenuButton<T>> {
   /// show the menu of the button with `globalKey.currentState.showButtonMenu`.
   void showButtonMenu() {
     final PopupMenuThemeData popupMenuTheme = PopupMenuTheme.of(context);
-    final RenderBox button = context.findRenderObject()! as RenderBox;
-    final RenderBox overlay = Navigator.of(context).overlay!.context.findRenderObject()! as RenderBox;
-    final RelativeRect position = RelativeRect.fromRect(
-      Rect.fromPoints(
-        button.localToGlobal(widget.offset, ancestor: overlay),
-        button.localToGlobal(button.size.bottomRight(Offset.zero) + widget.offset, ancestor: overlay),
-      ),
-      Offset.zero & overlay.size,
-    );
     final List<PopupMenuEntry<T>> items = widget.itemBuilder(context);
+    // It is possible that the fade-out animation of the menu has not finished
```

Comment:
To make sure this doesn't regress, we should add a test to make sure that mid-animation the crash doesn't happen

Reply:
I found this issue when I run the test for my patch.